src/main/java/frc/robot/vision/VisionFilters.java

```diff
@@ -0,0 +1,34 @@
+package frc.robot.vision;
+
+import edu.wpi.first.math.geometry.*;
+import frc.robot.vision.filters.Filter;
+import frc.robot.vision.rawdata.RawAprilTagVisionData;
+import frc.robot.vision.rawdata.RawVisionData;
+
+public class VisionFilters {
+
+	protected static Filter<RawVisionData> isPitchInTolerance(Rotation2d pitchTolerance) {
+		return new Filter<>(rawVisionData -> Math.abs(rawVisionData.getEstimatedPose().getRotation().getY()) <= pitchTolerance.getRadians());
+	}
+
+	protected static Filter<RawVisionData> isRollInTolerance(Rotation2d rollTolerance) {
+		return new Filter<>(rawVisionData -> Math.abs(rawVisionData.getEstimatedPose().getRotation().getX()) <= rollTolerance.getRadians());
+	}
+
+	protected static Filter<RawVisionData> isRobotOnGround(double robotToGroundToleranceMeters) {
+		return new Filter<>(rawVisionData -> rawVisionData.getEstimatedPose().getZ() <= robotToGroundToleranceMeters);
+	}
+
+	//@formatter:off
+	protected static Filter<RawAprilTagVisionData> isAprilTagHeightInTolerance(
```

Comment:
The latest version of the formatter does a fine job with it. The sig will be inline but overall you can remove the formatter:off/on comments.